rest/config_startup.go

```diff
@@ -102,13 +103,16 @@ type BootstrapConfig struct {
 }
 
 type APIConfig struct {
-	PublicInterface  string `json:"public_interface,omitempty"  help:"Network interface to bind public API to"`
-	AdminInterface   string `json:"admin_interface,omitempty"   help:"Network interface to bind admin API to"`
-	MetricsInterface string `json:"metrics_interface,omitempty" help:"Network interface to bind metrics API to"`
-	ProfileInterface string `json:"profile_interface,omitempty" help:"Network interface to bind profiling API to"`
+	PublicInterface     string `json:"public_interface,omitempty"  help:"Network interface to bind public API to"`
+	AdminInterface      string `json:"admin_interface,omitempty"   help:"Network interface to bind admin API to"`
+	MetricsInterface    string `json:"metrics_interface,omitempty" help:"Network interface to bind metrics API to"`
+	DiagnosticInterface string `json:"diagnostic_interface,omitempty" help:"Network interface to bind diagnostic API to"`
+	ProfileInterface    string `json:"profile_interface,omitempty" help:"Network interface to bind profiling API to"`
+
+	AdminInterfaceAuthentication      *bool `json:"admin_interface_authentication,omitempty" help:"Whether the admin API requires authentication"`
+	MetricsInterfaceAuthentication    *bool `json:"metrics_interface_authentication,omitempty" help:"Whether the metrics API requires authentication"`
+	DiagnosticInterfaceAuthentication *bool `json:"diagnostic_interface_authentication,omitempty" help:"Whether the metrics API requires authentication"`
```

Comment:
I think there's a good argument for not being able to disable authentication for this in general, because either it is going to be publicly routable, or it will require adminPrivs.

Once we add a new config feature we can't remove it.